### PR TITLE
 Improve FileDialog filters

### LINKED ISSUE #18534: File Dialog's returned path does not account for suffix filter correctly
I could not find a duplicate, but I find it hard to believe this hasn't been encountered before. Then again most users are building games, not applications that might use this feature.

**Godot version:** 3.0.2

**OS/device including version:**
Linux Mint 17.3

**Issue description:**
If you include a suffix filter (simply called 'filter') to a FileDialog, it will allow the user to type in the name of the file they want to save and the suffix filter will be appended to the end of the file. As an example, let's say a file name is 'mydocument' in the file naming box and the filter is set to '.godoc'. The result _should_ be something like '/home/agameraaron/Documents/mydocument.godoc' that should be returned as the path. Instead, it misses the dot / period. It currently outputs to '/home/agameraaron/Documents/**mydocumentgodoc**'
No dot to separate the file suffix.

The consequence being is my program is slightly less fool proof if I disable filters as users can name to whatever format they want, yet only load in a certain type. If I keep it enabled, I'd have to make a small hack/workaround that just inserts the dot (which is what I'll do).

**Steps to reproduce:**
Use the file_selected signal on a FileDialog node. Use this code:
```
func _on_FileDialog_file_selected(path):
	print(path)
```
Activate the node by showing it, then save a file (it won't save anything on it's own, or at least I don't think it's supposed to and doesn't). It will output to the screen the offending result.

### PR DESCRIPTION
**Pull request by KoBeWi**
Resolves #18534

### CODE REVIEW

File: scene/gui/file_dialog.cpp [hunk 1]
```diff
@@ -649,6 +649,7 @@ void FileDialog::clear_filters() {
 }
 
 void FileDialog::add_filter(const String &p_filter) {
+	ERR_FAIL_COND_MSG(p_filter.begins_with("."), "Filter must be \"filename.extension\", can't start with dot.");
```
Comment (akien-mga) [inline scene/gui/file_dialog.cpp:652]: Would there be a possibility that users would want to filter on hidden files starting with `.`?

Comment (KoBeWi) [reply scene/gui/file_dialog.cpp:652]: I think it's rather corner-case. And as I commented in the issue, filters starting with `.` don't work as expected. Supporting hidden files would require fixing that.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.5.

**Comment by akien-mga** [comment]
Cherry-picked for 3.4.3.

**Pull request merged**